htlcswitch/link.go

```diff
@@ -1912,6 +1912,25 @@ func (l *channelLink) HtlcSatifiesPolicy(payHash [32]byte,
 		return failure
 	}
 
+	// Next, ensure that the passed HTLC isn't too large. If so, we'll cancel
+	// the HTLC directly.
+	if policy.MaxHTLC != 0 && amtToForward > policy.MaxHTLC {
```

Comment:
This is also a restriction that is questionable... Currently we  do allow forwarding on disabled channels for instance. Whether we should allow it or not is a discussion in itself, and I think it warrants a separate PR :)

Reply:
Submitted as a separate PR n #2436